cfn-guard/src/parser.rs

```diff
@@ -103,6 +107,28 @@ pub(crate) fn parse_rules(
                 debug!("Line is white space");
                 continue;
             }
+            LineType::Import => {
+                let import_cap = IMPORT_REG.captures(trimmed_line).unwrap();
+                let import_file = match fs::canonicalize(&import_cap["ruleset_path"]) {
+                    Ok(canonicalized_file) => canonicalized_file,
+                    Err(e) => return Err(e.to_string())
+                };
+
+                if imported_files.contains(&import_file) {
```

Comment:
currently this would just disallow duplicate imports of files. if ruleset a imported ruleset b and ruleset c, which both imported ruleset d, it would fail. might need to be a bit more sophisticated in checking for circular imports

Reply:
maybe we can soft fail and ignore circular imports. if something is already imported ignore?